libtelemetry/src/full/java/com/mapbox/android/telemetry/provider/MapboxTelemetryInitProvider.java

```diff
@@ -36,14 +42,30 @@ public boolean onCreate() {
         // Install crash reporter for telemetry packages only!
         MapboxUncaughtExceptionHanlder.install(getContext(), MAPBOX_TELEMETRY_PACKAGE, BuildConfig.VERSION_NAME);
       }
-      LocationCollectionClient.install(getContext(), TimeUnit.HOURS.toMillis(DEFAULT_SESSION_ROTATION_INTERVAL_HOURS));
+      initLocationCollectionClient();
     } catch (Throwable throwable) {
       // TODO: log silent crash
       Log.e(TAG, throwable.toString());
     }
     return false;
   }
 
+  private void initLocationCollectionClient() {
+    int initDelay = TelemetryUtils.obtainInitDelay(getContext());
+
+    handler.postDelayed(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          LocationCollectionClient.install(getContext(),
+            TimeUnit.HOURS.toMillis(DEFAULT_SESSION_ROTATION_INTERVAL_HOURS));
+        } catch (Exception exception) {
+          Log.e(TAG, exception.getMessage());
+        }
+      }
+    }, initDelay);
```

Comment:
I'm not sure if I understand the solution here. We will solely delay the initialization, it will still be executed on the main thread?